### DISCUSSION TITLE
How to load static files on `/server/api/` functions?

### DISCUSSION THREAD

**Discussion by cosbgn** [nuxt/nuxt#19860] [Questions]
I have a bunch of .json files which are several MB and I don't want to store them in a database.

What's the easiest approach to be able to read them? Currently I store them on `/server/files/example.json` and load them like this: 

```js
export default defineEventHandler(async (event) => {
    const db = JSON.parse(readFileSync(new URL(`../../server/files/example.json`, import.meta.url), 'utf8'))
```

This works fine in `dev` but when I deploy to vercel it can't find the path. Is there a way to store static files (like `/public` for the frontend) but which works with the backend (`/server`) ?

**Comment by wobsoriano** [comment]
You want to import it so it gets included in the prod build.

```js
import exampleJson from '@/server/files/example.json'
```

Something like that.

**Reply by mbledkowski** [reply]
Ok, how do you for example import toml file?

**Reply by mbledkowski** [reply]
I mean, what modifications to build system are needed, if possible at all

**Comment by TheAlexLichter** [comment]
You can also use [`useStorage`](https://nitro.unjs.io/guide/storage) via https://github.com/unjs/unstorage/ ☺️

**Reply by codeflorist** [reply]
For anyone coming across this, here is how this currently works:

Here is the code to access `/server/assets/example.json`:

```js
const assetStorage = useStorage('assets:server')
const exampleJson = await assetStorage.getItem('example.json')
return exampleJson
```

Basically one should place these files in `/server/assets`, since it is available automatically. Other folders must be made available via config (see https://nitro.build/guide/assets#server-assets).

**Reply by codeflorist** [reply]
Hmm, so, `useStorage` seems to be a viable way to get the contents of a file. But what if i need the actual path to the file (e.g. to state path to a font to be used to a pdf-library?

Stating `server/assets/fonts/OpenSans-Regular.ttf` works in dev mode, but on node-server it throws the error, that it can't fine the file at `<project-folder>\.output\server\assets\fonts\OpenSans-Regular.ttf`

**Comment by cosbgn** [comment]
Yes I think the correct way is `useStorage` so it will work anywhere. Just make sure you set the path correctly; look at this: https://github.com/unjs/nitro/issues/914
